### ISSUE TITLE
use tini in docker entry points

### ISSUE THREAD

**Issue by shankarkc** [SeleniumHQ/docker-selenium#330]
## Meta -
Image(s):  

Both
Docker Version:  

All
## Expected Behavior -
Use tini to handle crashed defunct process. (https://github.com/krallin/tini)
## Actual Behavior -
Docker image should be able to handle zombie containers. It has been observed that many times browser crash result in defunct process and containers become zombie. Docker Jenkins uses it to avoid zombie containers/process. While running our tests we have encountered zombie containers. After using tini the stopped appearing.

**Comment by diemol** [comment]
This is actually a good idea @shankarkc, perhaps you can consider to submit a PR with a proposed change?

**Comment by ddavison** [comment]
super idea 👍   since no progress has been made on this, i'll go ahead and start with the change

**Comment by ddavison** [comment]
note:  

> NOTE: If you are using Docker 1.13 or greater, Tini is included in Docker itself. This includes all versions of Docker CE. To enable Tini, just pass the --init flag to docker run.

[source](https://github.com/krallin/tini#using-tini)

that's nice!  let's make sure to document that

**Comment by lcnja** [comment]
Let me try
[#734](https://github.com/SeleniumHQ/docker-selenium/pull/734)

**Comment by diemol** [comment]
A proof of concept shows that `tini` is already included in docker since 1.13 and working properly:

Without `--init` (process ID 1 is the entry point):
```
docker run -d -p 4444:4444 --name selenium-hub selenium/hub:3.12.0-americium

docker exec -ti selenium-hub bash

seluser@b2e764c4f3a4:/$ ps aux | cat
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
seluser      1  0.1  0.0  18036  2932 ?        Ss   09:59   0:00 /bin/bash /opt/bin/entry_point.sh
seluser     10  6.0  0.8 5154544 69404 ?       Sl   09:59   0:01 java -jar /opt/selenium/selenium-server-standalone.jar -role hub -hubConfig /opt/selenium/config.json
seluser     35  0.1  0.0  18236  3216 pts/0    Ss   10:00   0:00 bash
seluser     44  0.0  0.0  34420  2804 pts/0    R+   10:00   0:00 ps aux
seluser     45  0.0  0.0   4528   664 pts/0    S+   10:00   0:00 cat

``` 

Using `--init` (process ID 1 is `/dev/init`):

```
docker run -d --init -p 4444:4444 --name selenium-hub selenium/hub:3.12.0-americium

docker exec -ti selenium-hub bash

seluser@bcf8700349bb:/$ ps aux | cat
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
seluser      1  0.0  0.0   1044     4 ?        Ss   09:52   0:00 /dev/init -- /opt/bin/entry_point.sh
seluser      7  0.0  0.0  18036  2848 ?        S    09:52   0:00 /bin/bash /opt/bin/entry_point.sh
seluser     11  0.7  0.8 5154544 68828 ?       Sl   09:52   0:02 java -jar /opt/selenium/selenium-server-standalone.jar -role hub -hubConfig /opt/selenium/config.json
seluser     36  0.0  0.0  18236  3276 pts/0    Ss   09:52   0:00 bash
seluser     46  0.0  0.0  34420  2804 pts/0    R+   09:57   0:00 ps aux
seluser     47  0.0  0.0  18236   480 pts/0    R+   09:57   0:00 bash

```

Considering that this issue has been almost completely stale over several months and the `tini` part is documented by docker and tini itself, I think we don't need to do more here. If more interest comes to this, I think we could basically add the previous example in the README.
